### PR TITLE
Planning: save pull_over_info output from PathBoundDecider into plann…

### PR DESCRIPTION
**Pull request by Capri2014**
…ing_context instead of into frame

### CODE REVIEW

File: modules/planning/tasks/deciders/path_bounds_decider.cc [hunk 1]
```diff
@@ -364,8 +380,8 @@ bool PathBoundsDecider::SearchPullOverPosition(
       has_a_feasible_window = true;
       const auto& reference_line = reference_line_info.reference_line();
 
-      double pull_over_s = std::get<0>(path_bound[(i+j)/2]);
-      double pull_over_l = std::get<1>(path_bound[(i+j)/2]);
+      const double pull_over_s = std::get<0>(path_bound[(i + j) / 2]);
+      const double pull_over_l = std::get<1>(path_bound[(i + j) / 2]);
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/path_bounds_decider.cc]: Save `path_bound[(i + j) / 2]` to a variable and reuse.

File: modules/planning/tasks/deciders/path_bounds_decider.h [hunk 2]
```diff
@@ -114,7 +114,8 @@ class PathBoundsDecider : public Decider {
   bool SearchPullOverPosition(
       const ReferenceLineInfo& reference_line_info,
       const std::vector<std::tuple<double, double, double>>& path_bound,
-      std::tuple<double, double, double>* const pull_over_configuration);
+      std::tuple<double, double, double, double, double>* const
```
Comment (xiaoxq) [inline modules/planning/tasks/deciders/path_bounds_decider.h]: Long tuples are hard to read and maintain. Use struct or proto instead.
But it's OK leave as it is for this PR.

Comment (Capri2014) [reply modules/planning/tasks/deciders/path_bounds_decider.h]: Yes, that is why I left a TODO for me and jiacheng for a re-factor

File: modules/planning/tasks/deciders/path_bounds_decider.cc (hunk 1 above)
Comment (Capri2014) [reply modules/planning/tasks/deciders/path_bounds_decider.cc]: Done

**Pull request merged**